cmd/ts/ping.go

```diff
@@ -0,0 +1,244 @@
+package ts
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/netip"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/loft-sh/devpod/pkg/tailscale"
+
+	"github.com/spf13/cobra"
+	ts "tailscale.com/client/tailscale"
+	"tailscale.com/ipn/ipnstate"
+	"tailscale.com/tailcfg"
+)
+
+type PingCmd struct {
+	*TsNetFlags
+
+	num         int
+	size        int
+	untilDirect bool
+	verbose     bool
+	tsmp        bool
+	icmp        bool
+	peerAPI     bool
+	timeout     time.Duration
+}
+
+func NewPingCmd() *cobra.Command {
+	cmd := &PingCmd{TsNetFlags: &TsNetFlags{}}
+	var pingCmd = &cobra.Command{
+		Use:     "ping",
+		Example: "tailscale ping <hostname-or-IP>",
```

Comment:
this is based on open source tailscale client implementation right? Can we put links to original files somewhere at the top?

Reply:
yup, I added a docblock to the package but I'll add the exact file at the top of each of the commands :+1: